src/hint_processor/builtin_hint_processor/secp/bigint_utils.zig

```diff
@@ -0,0 +1,413 @@
+            var result = Felt252.zero();
+            inline for (0..NUM_LIMBS) |i| {
+                result = result + (self.limbs[i] << (i * 86));
+            }
+
+            return result;
+        }
+
+        pub fn split(self: *Self, num: Int) Self {
+            const limbs = splitBigInt(std.mem.Allocator, num, self.limbs.len, 128);
+            return self.fromValues(limbs);
+        }
+
+        // @TODO: implement from. It is dependent on split function.
+        pub fn from(self: *Self, value: Int) !Self {
+            // Assuming we have a `split` function in BigIntN.zig that performs the conversion.
+            return self.split(value);
+        }
+    };
+}
+
+// Implements hint:
+// %{
+//    from starkware.cairo.common.cairo_secp.secp_utils import split
+//    segments.write_arg(ids.res.address_, split(value))
+// %}
+///
+pub fn nondetBigInt3(vm: *CairoVM, exec_scopes: *ExecutionScopes, ids_data: std.StringHashMap(HintReference), ap_tracking: ApTracking) !void {
+    const res_reloc = try hint_utils.getRelocatableFromVarName("res", vm, ids_data, ap_tracking);
+    const value = try exec_scopes.getRef("value") catch return HintError.IdentifierHasNoMember;
+
+    const arg = bigInt3Split(value);
+    vm.segments.writeArg(std.ArrayList(BigInt), res_reloc, arg);
+}
+
+// Implements hint
+// %{ ids.low = (ids.x.d0 + ids.x.d1 * ids.BASE) & ((1 << 128) - 1) %}
+pub fn bigintToUint256(vm: *CairoVM, ids_data: std.StringHashMap(HintReference), ap_tracking: ApTracking, constants: std.StringHashMap(Felt252)) !void {
+    const x_struct = try hint_utils.getRelocatableFromVarName("x", vm, ids_data, ap_tracking);
+    const d0 = try vm.getFelt(x_struct);
+    const d1 = try vm.getFelt(x_struct.offset(1));
+    const base_86 = constants.get("BASE_86") orelse return HintError.MissingConstant;
+    const mask = pow2ConstNz(128);
+    const low = ((d0 + (d1 * base_86)) % mask);
+    try hint_utils.insertValueFromVarName(std.mem.Allocator, "low", low, vm, ids_data, ap_tracking);
+}
+
+// Implements hint
+// %{ ids.len_hi = max(ids.scalar_u.d2.bit_length(), ids.scalar_v.d2.bit_length())-1 %}
+pub fn hiMaxBitlen(vm: *CairoVM, ids_data: std.StringHashMap(HintReference), ap_tracking: ApTracking) !void {
+    var scalar_u: BigInt3 = undefined;
+    var scalar_v: BigInt3 = undefined;
+    scalar_u = try scalar_u.fromVarName("scalar_u", vm, ids_data, ap_tracking);
+    std.debug.print("🎉🎉🎉🎉🎉🎉  {} \n", .{scalar_u});
+    scalar_v = try scalar_v.fromVarName("scalar_v", vm, ids_data, ap_tracking);
+
+    // get number of bits in the highest limb
+    const len_hi_u = scalar_u.limbs[2].numBits();
+    const len_hi_v = scalar_v.limbs[2].numBits();
+
+    const len_hi = @max(len_hi_u, len_hi_v);
+
+    // equal to `len_hi.wrapping_sub(1)`
+    const res = if (len_hi == 0) Felt252.Max.toInteger() else len_hi - 1;
+
+    try hint_utils.insertValueFromVarName(std.testing.allocator, "len_hi", MaybeRelocatable.fromInt(u256, res), vm, ids_data, ap_tracking);
```

Comment:
Your error certainly comes from here: you explicitly use a testing allocator `std.testing.allocator` inside a general function and this is not allowed: testing allocator should only be used inside test blocks. To circumvent this issue, you can pass the allocator as a function argument of `hiMaxBitlen` and use the testing allocator to call your function inside the test block.